MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/annotations/ClickOnLayerActivity.java

```diff
@@ -0,0 +1,153 @@
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_click_on_layer);
+
+    Toast.makeText(ClickOnLayerActivity.this, R.string.click_on_polygon_toast_instruction,
+        Toast.LENGTH_SHORT).show();
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+
+        try {
+
+          // Load GeoJSONSource
+          source = new GeoJsonSource(geoJsonSourceId, new URL("https://gist.githubusercontent"
+              + ".com/tobrun/cf0d689c8187d42ebe62757f6d0cf137/raw/4d8ac3c8333f1517df9d303"
+              + "d58f20f4a1d8841e8/regions.geojson"));
+
+          // Add GeoJsonSource to map
+          mapboxMap.addSource(source);
+
+        } catch (MalformedURLException malformedUrlException) {
+          Log.d("Invalid URL", "malformedUrlException = " + malformedUrlException.toString());
+        }
+
+        // Create FillLayer with GeoJSON source and add the FillLayer to the map
+        layer = new FillLayer(geoJsonLayerId, geoJsonSourceId);
+        layer.setProperties(fillOpacity(0.5f));
+        mapboxMap.addLayer(layer);
+
+        mapboxMap.setOnMapClickListener(new MapboxMap.OnMapClickListener() {
+          @Override
+          public void onMapClick(@NonNull LatLng point) {
+            PointF pointf = mapboxMap.getProjection().toScreenLocation(point);
+            RectF rectF = new RectF(pointf.x - 10, pointf.y - 10, pointf.x + 10, pointf.y + 10);
+
+            List<Feature> featureList = mapboxMap.queryRenderedFeatures(rectF, geoJsonLayerId);
+
+            for (com.mapbox.services.commons.geojson.Feature feature : featureList) {
+              Log.d("Feature found with %1$s", feature.toJson());
+
+              Toast.makeText(ClickOnLayerActivity.this, R.string.click_on_polygon_toast,
+                  Toast.LENGTH_SHORT).show();
+
+            }
+          }
+        });
+
+      }
+    });
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
```

Comment:
Have added `android:launchMode="singleTop"` to manifest file to replace `onOptionsItemSelected`